src/clj/fluree/db/connection/system.cljc

```diff
@@ -0,0 +1,159 @@
+
+(defmethod ig/init-key :fluree.storage/file
+  [_ storage-path]
+  (file-store/open storage-path))
+
+(defmethod ig/init-key :fluree.storage/memory
+  [_ _]
+  (memory-store/create))
+
+#?(:clj
+   (defmethod ig/init-key :fluree.storage/s3
+     [_ {:keys [bucket prefix endpoint]}]
+     (s3-store/open bucket prefix endpoint)))
+
+(defmethod ig/init-key :fluree.storage/ipfs
+  [_ endpoint]
+  (ipfs-store/open endpoint))
+
+(defmethod ig/init-key :fluree.storage/remote-resource
+  [_ servers]
+  (remote-store/resource servers))
+
+#?(:cljs (defmethod ig/init-key :fluree.storage/localstorage
+           [_ _]
+           (localstorage-store/open)))
+
+(defmethod ig/init-key :fluree.nameservice/storage-backed
+  [_ {:keys [address-prefix store]}]
+  (storage-nameservice/start address-prefix store))
+
+(defmethod ig/init-key :fluree.nameservice/ipns
+  [_ {:keys [server profile]}]
+  (ipns-nameservice/initialize server profile))
+
+(defmethod ig/init-key :fluree.serializer/json
+  [_ _]
+  (json-serde))
+
+(defmethod ig/init-key :fluree.connection/id
+  [_ _]
+  (str (random-uuid)))
+
+(defmethod ig/init-key :fluree.connection/state
+  [_ _]
+  (connection/blank-state))
+
+(defmethod ig/init-key :fluree.index/storage
+  [_ {:keys [storage serializer cache]}]
+  (index.storage/index-store storage serializer cache))
+
+(defmethod ig/init-key :fluree/connection
+  [_ config]
+  (connection/connect config))
+
+(defn base-config
+  [parallelism cache-max-mb defaults]
+  {:fluree.serializer/json            {}
+   :fluree/cache                      cache-max-mb
+   :fluree.nameservice/storage-backed {:store (ig/ref :fluree/byte-storage)}
+   :fluree.index/storage              {:storage    (ig/ref :fluree/content-storage)
+                                       :serializer (ig/ref :fluree/serializer)
+                                       :cache      (ig/ref :fluree/cache)}
+   :fluree.connection/id              {}
+   :fluree.connection/state           {}
+   :fluree/connection                 {:id          (ig/ref :fluree.connection/id)
+                                       :state       (ig/ref :fluree.connection/state)
+                                       :cache       (ig/ref :fluree/cache)
+                                       :store       (ig/ref :fluree/content-storage)
+                                       :index-store (ig/ref :fluree.index/storage)
+                                       :serializer  (ig/ref :fluree/serializer)
+                                       :primary-ns  (ig/ref :fluree/nameservice)
+                                       :aux-nses    []
+                                       :parallelism parallelism
+                                       :defaults    defaults}})
+
+(defn memory-config
+  [parallelism cache-max-mb defaults]
+  (-> (base-config parallelism cache-max-mb defaults)
+      (assoc :fluree.storage/memory {})
+      (assoc-in [:fluree.nameservice/storage-backed :address-prefix] "fluree:memory://")))
```

Comment:
This is a personal bugaboo, so feel free to not take it seriously, but I would really love to see the double-slash `//` removed. Tim Berners-Lee calls the decision to add the double slash his [chief regret](https://archive.nytimes.com/bits.blogs.nytimes.com/2009/10/12/the-webs-inventor-regrets-one-small-thing/), and I'd really love to not perpetuate it : )

Reply:
The intention here has always been that this allows us to distinguish a location. it is probably misused a bit here with 'memory' as doesn't really make sense in this situation.

When a // exists, we know a remote location is being specified (e.g. IP/dns address, but it could be a P2P network address or otherwise).

I've looked into multi-addr as an alternative, but it has more complexity than I think we'll ever need and it is something new people have to learn. For that reason I have preferred the // as, like it or not, it is an established convention.